README.md

```diff
@@ -49,16 +85,211 @@ grey_fg2 = 20% lighter than grey
+| `standout`      | decorations                                                        | `boolean` | boolean value                               |
+| `underline`     | decorations                                                        | `boolean` | boolean value                               |
+| `undercurl`     | decorations                                                        | `boolean` | boolean value                               |
+| `funderdouble`  | decorations                                                        | `boolean` | boolean value                               |
+| `underdotted`   | decorations                                                        | `boolean` | boolean value                               |
+| `underdashed`   | decorations                                                        | `boolean` | boolean value                               |
+| `strikethrough` | decorations                                                        | `boolean` | boolean value                               |
+| `reverse`       | decorations                                                        | `boolean` | boolean value                               |
+| `nocombine`     | decorations                                                        | `boolean` | boolean value                               |
+| `default`       | Don't override existing definition if true                         | `boolean` | boolean value                               |
+| `ctermfg`       | number Sets foreground of cterm color                              | `number`  | number value                                |
+| `ctermbg`       | number Sets background of cterm color                              | `number`  | number value                                |
+| `cterm`         | comma-separated list of cterm opts. (more info `:h highlightings`) | `string`  | string value                                |
+
+- note that `fg`, `bg`, `link`, `italic`, `bold` are common used
+
+```lua
+["@variable"] = { fg = "#E4F0FB" },
+Comment = { bg = "yellow", italic = true },
+["@string"] = { link = "String" }
+```
+
+- see [Tips](#tips) that will help you
+
+## Contribute and test your theme
+
+see how themes being created [here](#examples)
+
+### testing
+
+- you can test your plugin by adding your file in
+  `custom/themes/<YOUR FILE NAME>.lua`
+- and add your theme name
+
+```lua
+M.require("base46").override_theme(M, "<YOUR THEME NAME>")
+```
+
+then go to `custom/chadrc.lua` and set your theme
+
+```lua
+M.ui = {
+ theme = "<YOUR THEME NAME>"
+}
+```
+
+## tips
+
+- use a highcontrastcolor e.g: yellow or red and run neovim command `:hi` to
+  figure out what highlightings uses this color
+- install the `nvim-treesitter/playground` plugin to help you inspect
+  highlightings are used
+- use the `:Inspect` or use `:TSCaptureUnderCursor` commands (required
```

Comment:
`:Inspect` and `:InspectTree` are in neovim, not necessarily needing of `nvim-treesitter/playground`. Not to mention `:TSCaptureUnderCursor` doesn't show all the hlgroups that's responsible for the cursor position. Better to not mention it at all